src/main.rs

```diff
@@ -85,70 +86,70 @@ async fn main() {
+        (Some(org_name), _) | (None, Some(org_name)) => org_name,
+        (None, None) => panic!("no org name given"),
+    };
+
+    let num_days: i64 = match config.github.number_of_days {
+        Some(n) => n.try_into().unwrap(),
+        None => 30,
+    };
+
+    let repo: Option<String> = match (cli.repo, config.github.repo) {
+        (Some(repo_name), _) | (None, Some(repo_name)) => Some(repo_name),
+        (None, None) => None,
+    };
+
+    let sheet_id: Option<String> = match (cli.google_sheet, config.google_sheet) {
+        (Some(s_id), _) | (None, Some(s_id)) => Some(String::from(s_id)),
+        (None, None) => None,
+    };
+
+    let mut column_names: Option<Vec<String>> = None;
+    let (tx, mut rx) = mpsc::channel::<Vec<String>>(400);
+
+    if let Some(cmd) = cli.cmd {
+        match cmd {
+            Cmd::List => {
+                let list_repos = ListReposForOrg::new(org_name, num_days);
+                column_names = Some(list_repos.column_names());
+                println!("{:#?}", &list_repos);
+
+                tokio::spawn(async move {
+                    if let Err(e) = list_repos.producer_task(tx).await {
+                        println!("Encountered an error while collecting data: {:#?}", e);
+                    };
+                });
+            }
+            Cmd::RepoParticipants => {
+                let repo_participants = RepoParticipants::new(org_name, repo, 30);
+                column_names = Some(repo_participants.column_names());
+
+                tokio::spawn(async move {
+                    if let Err(e) = repo_participants.producer_task(tx).await {
+                        println!("Encountered an error while collecting data: {}", e);
+                    };
+                });
             }
         }
+    }
 
-        Opt::RepoParticipants {
-            org,
-            repo,
-            verbose: _,
-        } => {
-            let (tx, mut rx) = mpsc::channel::<Vec<String>>(400);
-            let repo_participants = RepoParticipants::new(org, repo, 30);
-            let column_names = repo_participants.column_names();
-
-            tokio::spawn(async move {
-                if let Err(e) = repo_participants.producer_task(tx).await {
-                    println!("Encountered an error while collecting data: {}", e);
-                };
-            });
-
-            if let Err(e) = Print::consume(Print, &mut rx, column_names).await {
-                println!("Error while printing results: {}", e);
-            }
+    // if user specified a google sheet ID, they must want to export data to that sheet
+    if let Some(sheet_id) = sheet_id {
+        if let Err(e) = ExportToSheets::new(&sheet_id)
+            .consume(&mut rx, column_names.clone().unwrap())
+            .await
+        {
+            println!("Error exporting to sheets: {}", e);
+        }
+    }
+    // if user specified the print flag, they must want to print to terminal
+    if cli.print {
```

Comment:
```suggestion
    else if cli.print {
```

we only do one or the other, right?

Reply:
Actually, maybe we should set things up so that there is an intermediate layer, and you can expert in multiple ways?